OpenWeather/App/Search/SearchViewModel.swift

```diff
@@ -28,14 +29,20 @@ final class SearchViewModel: ObservableObject, ErrorHandling {
         self.errorHandler = errorHandler
     }
 
-    func getLocations(query: String) {
+    func getLocations() {
         task?.cancel()
 
         task = Task {
-            let result = try? await locationProviding.locations(for: query)
-
-            await MainActor.run {
-                self.result = result ?? []
+            try? await Task.withCheckedCancellation {
+                do {
+                    let result = try await locationProviding.locations(for: searchText)
+
+                    await MainActor.run {
+                        self.result = result
+                    }
```

Comment:
This is better than no cancellation.. but ideally I think we need to check for cancellation before and after this line:
```swift
let result = try await locationProviding.locations(for: searchText)
```

But I'm also no expect at cancellations.. But ideally we would want to cancel the task after we get the data as we don't want to update the UI with stale data..

Reply:
Oh my mistake, it should be like this.